controllers/progressivesync_controller.go

```diff
@@ -267,6 +289,41 @@ func (r *ProgressiveSyncReconciler) getClustersFromSelector(ctx context.Context,
 	return secrets, nil
 }
 
+// calculateHashValue returns a hash value string for the ApplicationSet spec
+func (r *ProgressiveSyncReconciler) calculateHashedSpec(ctx context.Context, ps *syncv1alpha1.ProgressiveSync) (string, error) {
+	key := client.ObjectKeyFromObject(ps)
+	latest := syncv1alpha1.ProgressiveSync{}
+	if err := r.Client.Get(ctx, key, &latest); err != nil {
+		r.Log.Error(err, "Failed to retrieve the ProgressiveSync object while calculating the hash value")
+		return "", err
+	}
+
+	apiMetadata := strings.Split(*latest.Spec.SourceRef.APIGroup, "/")
+	genericObj := unstructured.Unstructured{}
```

Comment:
Why do we use the unstructured here? We now that we're getting a reference to the applicationset and we're not supporting a generic `SourceRef`.
Maybe this generate a new discussion about why don't we call it `AppSetRef` instead :)

Reply:
Initially we did it by retrieving an ApplicationSet obj, but after discussing it with Nebojsa we agreed that it should be more generic and allow for other object to be used as well. Maybe we should discuss how generic we want it to be.

Reply:
I'm not 100% sure it should be generic since this project is called `applicationset-progressive-sync` :)

Reply:
As we discussed, we agreed to change it back to ApplicationSet.